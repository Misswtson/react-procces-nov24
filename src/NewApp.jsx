import PropTypes from 'prop-types';

export const NewApp = ( { title, subTitle, name } ) => {


    return (
<>
  <h1 data-testid="test-tittle"> { title } </h1>
  {/* comand shift slach  */} 
  {/* <code> {JSON.stringify( NewMessage)}<code/>*/}
    <p>{ subTitle }</p>
    <p>{ name }</p>

    </>
    );
}

NewApp.propTypes = {
title: PropTypes.string.isRequired,
subTitle: PropTypes.number
}

NewApp.defaultProps = {
    name:'Emma',
    title:'Hola, soy Emma',
    subTitle: 'Lo lograras, ya lo ver√†s!!'
}