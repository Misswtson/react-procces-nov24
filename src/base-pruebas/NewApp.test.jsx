import { getByText, render } from '@testing-library/react';
import { NewApp } from '../NewApp';


describe('pruebas en NewApp.jsx', () => {

    test('Debe hacer match con el snapshot', () => {

        const tittle ='Hola, soy Emma';
        const { container } = render( < NewApp tittle = { tittle }/> )
        expect( container ).toMatchSnapshot();
     })
    
    
     test('Debe hacer match con el snapshot', () => {

        const tittle ='Hola, soy Emma';
        const { container, getByText, getByTestId} = render( < NewApp tittle = { tittle }/> )
        expect( getByText ).toBeTruthy();

        //const h1 = container.querySelector('h1');
        //expect(h1.innerHTML).toContain( tittle );
        expect( getByTestId('test-tittle').innerHTML ).toContain( tittle );
     })

     test('Debe mostrar el subtitulo enviado por props', () => { 
        
        const tittle ='Hola, soy Emma';
        const subTitle = 'Lo lograras, ya lo ver√†s!!';
        const { getByText } = render( 
        < NewApp 
            tittle = { tittle }
            subTitle ={ subTitle }/> 
            );
        expect( getByText (subTitle) ).toBeTruthy();
    })
});
