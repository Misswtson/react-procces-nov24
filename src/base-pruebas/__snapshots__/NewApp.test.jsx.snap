// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en NewApp.jsx Debe hacer match con el snapshot 1`] = `
<div>
  <div
    data-testid="test-tittle"
  >
     
    Hola, soy Emma
     
  </div>
  <p>
    Lo lograras, ya lo ver√†s!!
  </p>
  <p>
    Emma
  </p>
</div>
`;
